### DISCUSSION TITLE
Set led color based on Alarm state

### DISCUSSION THREAD

**Discussion by erwinvana** [adonno/tagreader#147] [Ideas]
Hello, 

Is there a way to have the LED stay on in red or green beased on the sate of the alarmo?
I'm testing this tagreader to arm and disarm my Home Assistent Alarmo install.
I really like the option to have the led be green if disarmed and red when armed.

Can this be done and is there a Noob explanation on how to ?

Kind regards

Erwin

**Comment by jesserockz** [comment]
Yes this can be done. You can import the state of the alarm as a `text_sensor` with `platform: homeassistant` and then change the led based on that state.

**Reply by erwinvana** [reply]
Sorry for asking all these questions, but can you please explain where and how i need to do this? this my first ESP project. All of my HA is tasmota flashed or install ready copied from others. Keen to learn tough ;-)

**Reply by jesserockz** [reply]
Take a look at the [homeassistant text_sensor](https://esphome.io/components/text_sensor/homeassistant.html) for getting the alarm state into ESPHome. From there you can use the [automations of the text sensor](https://esphome.io/components/text_sensor/index.html#on-value) to actually do something when the value is changed like changing the light with an [automation](https://esphome.io/guides/automations.html#if-action)

The documentation is full of examples for automations etc. Feel free to reach out on [discord](https://discord.gg/KhAMKrd) where it is a lot easier and faster to help with specific code once you have started writing it.

**Comment by brintal** [comment]
I found a noob-friendly way.

1. Open the Tag Reader LED entity, turn the LED on and then change the color (e.g. set it to green) to whatever you want:
![image](https://user-images.githubusercontent.com/6397462/197355044-11675282-dff9-48b0-bb14-6c94f75b0f20.png)

2. After that you can create a new scene in HA which includes the Tag Reader LED entity (e.g. call the scene 'Alarm LED Green'). Repeat Steps 1 and 2 for all Colors you wanna use.

3. Use the created scenes within your automations, e.g. activate Scene 'Alarm LED Green' when the alarm is disarmed.

**Comment by Kangle66-HA** [comment]
Hi all,
do you managed your issue?
I want to try to do the same thing.
Maybe having a RGB LED on differents pin?
How you did it?
thanks.
